core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -6574,14 +6574,82 @@ public void testGetDate() throws Exception {
 
       rs.close();
 
-      final String sql = "update t2 set vals=? where id=?";
```

Comment:
Why are you dropping this case?
I remember I recently (calcite 1.15) fixed a problem with jsbc params and UPDATE.

Can't we keep this test and add a new one

Reply:
I am considering this a main issue of change. Before the patch former update query was planned in underlying jdbc schema as a subquery. After the patch - such query is not pushed to underying jdbc through jdbcrules logic, but stays in calcite. Calcite itself does not implement in a planner UDPATES - so it throws exception. 

I believe this part of a test was a little hack'y - it relied on the invalid behaviour of calcite planner.

Reply:
From the change it seems to me that we are losing one of the most common cases, that is an UPDATE by PRIMARY KEY.
Maybe I am missing something.

Anyway it would be better to add an explicit testcase which tells what it is expected to happen with the query you dropped from the test case.
WDYT?

Reply:
I will try to look at this next week. Maybe rewrite the mentioned test not to use the Connection.prepareStatement but some internals.

Reply:
@eolivelli  I've recreated the test case. But now rather based on PreparedStatement it uses directly planner and traversing through Rel nodes. 

The former test worked by hack - it was relying on that the whole query was pushed to jdbc schema, thus omitting the ling4j implementation. 

When PreparedStatement is being prepared - whole query is compiled and update clause is not supported. So the former test worked only because update was pushed to jdbc subschema. 

Current tests - checks the same query as former and validates whether dynamic parameters exists and are specified. But it uses for that traversing through rel nodes directly.

I hope this clarifies the thing.